### ISSUE TITLE
Can't resolve `graphql-ws` from `@graphiql/toolkit/esm/create-fetcher`

### ISSUE THREAD

**Issue by benjdlambert** [graphql/graphiql#2405]
Seems like the recent release of `@graphiql/toolkit` has broken some of our tests. Maybe due to a missing dependency?

```
Module not found: Error: Can't resolve 'graphql-ws' in '/tmp/backstage-e2e-Z2eQgF/test-app/node_modules/@graphiql/toolkit/esm/create-fetcher'

error Command failed with exit code 1.
```

From https://github.com/backstage/backstage/runs/6420680730?check_suite_focus=true

**Comment by thomasheyenbrock** [comment]
Hey @benjdlambert 👋 `graphql-ws` is currently listed as a peer dependency of `@graphiql/toolkit` (see [here](https://github.com/graphql/graphiql/blob/main/packages/graphiql-toolkit/package.json#L34)), so you need to add that extra dependency on your end. We don't really make this clear though in the [docs](https://github.com/graphql/graphiql/blob/main/packages/graphiql-toolkit/docs/create-fetcher.md) for `createFetcher`.

In general, I don't see why we shouldn't add this as a first-class dependency, we also include `meros` as dependency (which we use for consuming multipart responses) 🤔 cc @acao maybe you have more context on this

**Comment by barbalex** [comment]
This hit's me when upgrading graphiql from v1.8.8 to 1.8.10

**Comment by benjdlambert** [comment]
@thomasheyenbrock maybe it does make sense to add `graphql-ws` as a required dependency of the packages that consume it directly and don't have a nice fallback or handling if it's not installed. Wondering if some of the moving around of code from #2401 made the library a required dependency now, and maybe because of the monorepo setup that the repo has, these issues don't present themselves until installed as separate packages standalone as a lot of packages get hoisted and node really doesn't care how something found itself into `node_modules` providing it exists. We have pretty much the same setup with `workspaces` and `changesets` as this repo, and we've been bitten by it a few times and had to enforce lint rules to try and catch stuff like this in CI.

**Comment by simPod** [comment]
Yes, please. Yarn pnp requires it. Or I did not find a way to install 1.8.10. I even tried

```yaml
packageExtensions:
  "@graphiql/toolkit@*":
    peerDependencies:
      graphql-ws: "*"
```

1) It doesn't work anyway, I'm doing sth wrong mb
2) it would be only a workaround

**Comment by acao** [comment]
Great discussion here everyone!

what if we just seperated some of this functionality into two separate, additional packages `@graphiql/toolkit-ws` and `@graphiql/toolkit-ws-legacy` that can somehow be consumed as “plugin“s by createFetcher()?? It seems like the only way to prevent non-ws users from worrying about this while also making it easy to setup for ws users

**Comment by thomasheyenbrock** [comment]
I'll try to summarise to make sure we're aligned.

The **problem** is that there's a top-level import of `graphql-ws` in the `@graphiql/toolkit` package. The package `graphql-ws` does only list `graphql-ws` as peer dependency, which is intended. This leads to an error if a user tries to consume the `@graphiql/toolkit` package without also installing `graphql-ws` as dependency. In particular this is cumbersome for users who don't need a fetcher that supports subscriptions as we enforce installing an extra dependency that will never be used.

This is _not_ a problem for users that use the CDN bundle, as this includes `graphql-ws`.

To summarise possible solutions (I'll throw mine in there as well):
- "Separate" the current `@graphiql/toolkit` package into multiple ones. There would be a new package `@graphiql/toolkit-ws` that lists `graphql-ws` as dependency. (Proposed by @acao)
- Avoid the top-level import, i.e. only import `graphql-ws` dynamically when there is a `subscriptionUrl` being passed to `createGraphiQLFetcher`. We could even wrap this dynamic import with a try-catch and throw a nice error that educates about the missing peer dependency (proposed by @thomasheyenbrock)

Lmk if we're aligned on the problem and what you think about possible solutions.

**Comment by benjdlambert** [comment]
@thomasheyenbrock, thinking the dynamic import is probably nicer right? not sure theres a need for multiple packages if we can do it in transparent way.

**Comment by thomasheyenbrock** [comment]
@benjdlambert would you like to get a PR in for that change?

**Comment by thomasheyenbrock** [comment]
This was published today with `@graphiql/toolkit@0.5.0`!

**Comment by julpat** [comment]
The issue still exists. You did not remove the "top-level import" of peer dependency :-|

Iam using my own `fetch` method, but still getting the error of missing dependency.

![image](https://user-images.githubusercontent.com/6079287/225298545-03b267b2-9c78-4da5-8b7c-ac22c99ff3a0.png)

**Comment by palyvodaBoi** [comment]
Yeap, it still exists:

<img width="1129" alt="Can't resolve 'graphql-ws' warning" src="https://github.com/user-attachments/assets/531fcf76-88cb-4e22-9f1a-bd3f9aa94fb4">
